LogParser.groovy

```diff
@@ -1,511 +1,370 @@
-    // calculate_common_filters()
+    // -------------------------------------------------
     // Description :
-    //      Takes all the filters and keeps a count of how many times a filter has been seen
-    // -------------------------------------------------------------------------------------
-    void calculate_common_filters() {
-        int i, j = 0;
-        int len = this.logs.size()
+    //      Parses the new log format that contains Json
+    // Params :
+    //      line : the log line to parse
+    // -------------------------------------------------
+    private void parse_new_log_line(String[] line){
+        final int SEARCH_START = 12
+
+        Date date             = get_date_from_line(line)            // date and time of the log
+        String type           = get_type_from_line(line)            // specifies what kind of search
+        String tQuery         = ""                                  // temporary string for the query
+        List<String> tFilters = []                                  // temporary variable to hold filters
+        Boolean tFacet        = false                               // temporary variable for the facet
+        def tPage             = new Tuple2<Integer, Integer>(0,0)   // temporary variable for the page/offset
 
-        this.filterMap.put( (this.logs[0].get_filters()[0]), 1)
+        String workingLine = line.drop(SEARCH_START).join(" ")
+        def slurper = new JsonSlurper()
+        def log = slurper.parseText(workingLine)
 
-        for (i = 0; i < len; i++){
-            int filterLen = this.logs[i].get_filters().size()
+        if (log.containsKey("id")){
+            RequestLog tmp = new RequestLog(date, type, log.id)
+            this.requestLogs.add(tmp)
+            return
+        }
 
-            for (j = 0; j < filterLen; j++){
-                if ( !(this.filterMap.containsKey(this.logs[i].get_filters()[j])) ){
-                    this.filterMap.put( (this.logs[i].get_filters()[j]), 1)
-                }
-                else{
-                    this.filterMap[this.logs[i].get_filters()[j]] += 1
-                }
-            }
+        if (log.containsKey("queries")) {
+            tQuery = log.queries.value
         }
-        this.filterMap = this.filterMap.sort { -it.value }
-    }
 
+        if (log.containsKey("filters")) {
+            /* Uncomment this for the format to be in JSON
+            tFilters = log.filters.collect{it ->
+                    JsonOutput.toJson(it)}
+            */
+            tFilters = log.filters
+        }
 
+        if (log.containsKey("facets")){
+            tFacet = log.facets
+        }
 
-    void calculate_common_facets(){
+        if (log.containsKey("page")){
+            tPage = new Tuple2<Integer, Integer>(log.page.max, log.page.offset)
+        }
 
+        SearchLog tmp = new SearchLog(date, type, tQuery, tFilters, tFacet, tPage)
```

Comment:
This is interesting because the scope is small so the variable name isn't that significant, but the semantics of the variable name 'tmp' doesn't match the actually object. It is a small change, but the variable should be named 'searchLog'. I think this is also preferable because searching for 'searchLog' is more intuitive than searching for 'tmp'.